core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala

```diff
@@ -526,6 +548,8 @@ class BlockManagerMasterEndpoint(
 
       blockManagerInfo(id) = new BlockManagerInfo(id, System.currentTimeMillis(),
         maxOnHeapMemSize, maxOffHeapMemSize, storageEndpoint, externalShuffleServiceBlockStatus)
+
+      addMergerLocation(id)
```

Comment:
I'm thinking about only adding the merger location when we know there're no active executors on it. Since we'd prefer active executor locations first now, I think it's almost a redundant copy of the active executor locations normally.

The idea is we could add it when we find there're no active executors after `removeExecutor` and remove it when there're new executors register on the same host.

WDYT?